### PR TITLE
[SPARK-25589][SQL][TEST] Add BloomFilterBenchmark

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

This PR aims to add `BloomFilterBenchmark`. For ORC data source, Apache Spark has been supporting for a long time. For Parquet data source, it's expected to be added with next Parquet release update.

## How was this patch tested?

Manual.

```scala
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.execution.benchmark.BloomFilterBenchmark"
```

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by dongjoon-hyun** [comment]
Thanks, @HyukjinKwon !

**Comment by dongjoon-hyun** [comment]
Rebased to `master` due to https://github.com/apache/spark/pull/22599.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you for review and approval, @HyukjinKwon .

Merged to master.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/BloomFilterBenchmark.scala [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      }
+    }
+  }
+
+  private def readBenchmark(): Unit = {
+    withTempPath { dir =>
+      val path = dir.getCanonicalPath
+
+      df.write.orc(path + "/withoutBF")
+      df.write.option("orc.bloom.filter.columns", "value").orc(path + "/withBF")
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/BloomFilterBenchmark.scala]: I think we should also add this ORC feature to the SQL doc.

**Pull request closed without merging**